botocore/awsrequest.py

```diff
@@ -383,51 +476,34 @@ def __init__(self,
 
     def prepare(self):
         """Constructs a :class:`AWSPreparedRequest <AWSPreparedRequest>`."""
-        return AWSPreparedRequest(self)
+        return self._REQUEST_PREPARER.prepare(self)
 
     @property
     def body(self):
-        p = AWSPreparedRequest(self)
-        p.prepare_body(self.data)
-        if isinstance(p.body, six.text_type):
-            p.body = p.body.encode('utf-8')
-        return p.body
+        body = self.prepare().body
```

Comment:
So why do we bother with any of the other prep if we only use the body

Reply:
Looks like we got an issue related to this: https://github.com/boto/botocore/issues/1561

Reply:
I've opened another PR to address this: https://github.com/boto/botocore/pull/1566